Dev

dombo3 opened the pull request.

dombo3 committed:
create UserServiceController, Add Account Table to DB, Rename HeartbeatController

dombo3 committed:
rename heartbeatcontroller, create register endpoint

dombo3 committed:
configure flyway plugin properties

dombo3 committed:
feat(migration/V3__Create_Account): create Account table with Version 3 flyway migration

dombo3 committed:
create model objects (Account, Attributes, Data, RequestData): parse JSON input at endpoint /register

dombo3 committed:
update /register endpoing: create new Account with posted informations

dombo3 committed:
create repository/AccountRepository, modify migration/V3__Create_Account: add admin remove password

dombo3 committed:
add JJWT dependency

dombo3 committed:
feat(register): save new user to database with a sample token. Response(JSON) with the user attributes from database: response with 201 status. (needs to refact, hard coded, meets user story criteria)

dombo3 committed:
create model structure based on JSON API (and given user story) JSON structure.

dombo3 committed:
Create AccountRepository to get user infos from DB (id,admin,email,token);

dombo3 committed:
Merge branch 'dev' into register

vvarro committed:
Merge pull request #27 from greenfox-academy/register

Register

zsiga112 committed:
chore(guardedTest): test ready

zsiga112 committed:
chore(Error): add Error model

zsiga112 committed:
chore(Response): add Error container

zsiga112 committed:
feat(guardedEnd): add new endpoint

dombo3 committed:
Merge branch 'dev' into jwt

dombo3 committed:
refact(register): outsource JWT creating into service/JwtCreator;

dombo3 committed:
Merge pull request #32 from greenfox-academy/jwt

Jwt

dombo3 committed:
refactor

dombo3 committed:
heartbeat package refactor

dombo3 committed:
logging package refactor

dombo3 committed:
rabbitmq package refact

dombo3 committed:
refactor package yondu

dombo3 committed:
register package refactor

dombo3 committed:
save encrypted password into the database using bcrypt

dombo3 committed:
feat(password:encryption):set log rounds in password encryption changable through env var

vvarro committed:
add aop dependencies and create heartbeataspect

vvarro committed:
change aop structure

dombo3 committed:
logging in user with valid credentials

zsiga112 committed:
Merge branch 'dev' into guardedEnd

zsiga112 committed:
refactor(guarded-test): test separated from others

zsiga112 committed:
chore(guarded-test): test ready

with valid token and provided header

zsiga112 committed:
feat(guarded-tests): test with invlalid token

zsiga112 committed:
refactor(guardian): adopt to the new project struc

dombo3 committed:
refact(login,register): (login) authenticate already registered user, (register) block already registered user with HttpStatus.Conflict;

dombo3 committed:
Merge pull request #34 from greenfox-academy/passwordAndLogin

Password and login

zsiga112 committed:
fix(tests): correct expected header value

zsiga112 committed:
feat(guarded): test pass with invalid token

zsiga112 committed:
feat(guarded): pass test without request header

zsiga112 committed:
feat(guardedEnd): test with valid token pass

dombo3 committed:
Merge pull request #35 from greenfox-academy/guardedEnd

Guarded end

dombo3 committed:
refact(Account): add constructor;

vvarro committed:
create LoggingAspect

dombo3 committed:
refactor(login/registration): create method for parsing Json input with gson

dombo3 committed:
refactor(register/login): create method for creating Json object with Gson

vvarro committed:
refactor rabbitmq consume method

dombo3 committed:
test(UserServiceControllerTest): build up test environment

dombo3 committed:
Merge branch 'gson' into test

dombo3 committed:
test(register, login): write integration tests for register and login endpoint for new, already registered and authorized, unauthorized user

dombo3 committed:
refact(register,login): fix logical typo NOT, outsource JSON parsing and creating

vvarro committed:
add error endpoint logging function

dombo3 committed:
refact(login,register): outsource gson handling and authorization into services, refact code structure

dombo3 committed:
feat(register,login): create Credentials for parsing json email and password fields;

dombo3 committed:
feat(register,login): create Services for authentication and json handling

dombo3 committed:
style(test): remove unecessary inports

dombo3 committed:
Merge pull request #37 from greenfox-academy/test

Test and Login Register controller Refactoring

vvarro committed:
set error endpoint logging function

drexeverest commented on src/main/java/com/greenfox/guardian/controller/GuardedEnd.java:
```diff
@@ -0,0 +1,41 @@
+public class GuardedEnd {
+
+  @GetMapping("/user/{id}")
+  public ResponseEntity getUserRelatedStuff(
```
Think about how this RestController could be extended, and depending on that refactor this endpoint

drexeverest commented on src/main/java/com/greenfox/register/service/AuthService.java:
```diff
@@ -0,0 +1,36 @@
+    } else {
+      account = accountRepository.findAccountByEmail(email);
+    }
+    if (!checkPassword(password, account.getPassword())) {
```
Be consistent, if checkAccount true has an invalid meaning, make checkPassword true an invalid meaning also

dombo3 replied:
Thanks, I am refactoring it.

drexeverest commented on src/main/java/com/greenfox/register/service/GsonService.java:
```diff
@@ -0,0 +1,37 @@
+    data.add("attributes", attributes);
+    attributes.addProperty("id", id.toString());
+    attributes.addProperty("email",email);
+    System.out.println(email);
```
Why sout?

dombo3 replied:
Yap. ok thanks, left over

drexeverest commented on src/main/java/com/greenfox/yondu/Yondu.java:
```diff
@@ -1,4 +1,4 @@
-package com.greenfox.model;
```
But why?

drexeverest commented on src/main/java/com/greenfox/yondu/YonduController.java:
```diff
@@ -1,17 +1,17 @@
-package com.greenfox.controller;
```
Take it more serious

dombo3 replied:
Come oooon!!!!

vvarro committed:
set aop endpoint logging in 'info' level

vvarro committed:
set aop endpoint logging in 'error' level

vvarro committed:
refactor loggingAspect

vvarro committed:
refactor LoggingAspect

dombo3 committed:
Merge branch 'dev' into aop

dombo3 committed:
Merge pull request #38 from greenfox-academy/aop

Aop

dombo3 committed:
refact(userServiceController): refact controller:
 - make code more readable, enhance level of abstraction,
 - remove unecessary sout,
 - reverse logic of checkAccount method to be consistent with checkPassword method

dombo3 committed:
Merge branch 'dev' into refact

dombo3 committed:
Merge pull request #40 from greenfox-academy/refact

refact(userServiceController): refact controller:
